### PR TITLE
Project notification documents

### PR DESCRIPTION
**Pull request by dhlevi**
Adding document support to project notifications.

See ticket [EE-875](https://bcmines.atlassian.net/browse/EE-875)

### CODE REVIEW

File: src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  constructor(
+    private router: Router,
+    private snackBar: MatSnackBar
+  ) { }
+
+  ngOnInit() {
+    this.documents = this.data.data;
+    this.paginationData = this.data.paginationData;
+    this.activeLegislationYear = 2018;
```
Comment (KitArmstrong) [inline src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts]: Just confirming that this default year is correct?

File: src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  selectItem(item) {
+    item.checkbox = !item.checkbox;
+    let count = 0;
+    this.documents.map(doc => {
```
Comment (KitArmstrong) [inline src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts]: This should probably be a `forEach`

File: src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts (hunk 1 above)
Comment (dhlevi) [reply src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts]: Good question. I'll check

File: src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts (hunk 2 above)
Comment (dhlevi) [reply src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts]: Copy-paste from existing document handlers. I can update the new stuff

File: src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts (hunk 1 above)
Comment (dhlevi) [reply src/app/project-notification/documents/project-notification-document-table-rows/project-notification-document-table-rows.component.ts]: It is correct

File: src/app/project-notification/documents/upload/upload.component.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+    this.documents.map(doc => {
+      const formData = new FormData();
+      formData.append('upfile', doc.upfile);
+      formData.append('project', this.currentProject._id);
+      formData.append('documentFileName', doc.documentFileName);
+      formData.append('documentSource', 'PROJECT-NOTIFICATION');
+      formData.append('displayName', this.documents.length > 1 ? doc.documentFileName : this.myForm.value.displayName);
+      formData.append('dateUploaded', new Date().toISOString());
+      formData.append('datePosted', new Date().toISOString());
```
Comment (mtCarto) [inline src/app/project-notification/documents/upload/upload.component.ts]: I think we need to append legislationYear here, as well. Document uploads are failing for me due to an api validation error on that field.

Comment (dhlevi) [reply src/app/project-notification/documents/upload/upload.component.ts]: I've added back all the default formData fields, most set to null as we don't need them for project notifications.

**Pull request merged**